src/metabase/models/serialization/utils.clj

```diff
@@ -65,3 +68,152 @@
         (hydrate hydration-key)
         (get hydration-key)
         identity-hash)))
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                          Serialization Process                                                 |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+;;; Utilities for writing serialization functions.
+(defn remove-timestamps
+  "Removes any key that ends with _at."
+  [m]
+  (let [ats (filter #(.endsWith (str %) "_at") (keys m))]
+    (apply dissoc m ats)))
+
+;;; The end result of serialization is a directory tree of YAML files. Some entities are written to individual files
+;;; (most of them, eg. collections and fields) while others are consolidated into a single file for the entire set (eg.
+;;; settings).
+;;;
+;;; Some entities cannot be loaded until others are also loaded, these are its *dependencies*. For example, a Field
+;;; can't be loaded until its Table is defined, and the Table can't be loaded without the Database. These dependencies
+;;; are derived from the serialized form, and then loaded recursively in postorder.
+(defmulti serdes-dependencies
+  "Given an entity map AS DECODED FROM YAML, returns a (possibly empty) list of its dependencies, where each dependency
+  is represented by `(or entity_id identity-hash)` of the target entity.
+
+  This is mostly part of the deserialization process, so it's based on the serialized map and not on an IModel instance.
+
+  Default implementation returns an empty vector, so only models that have dependencies need to implement this."
+  ; TODO What actually is the right key for the entity type?
+  :name)
+
+(defmethod serdes-dependencies :default [_]
+  [])
+
+(p.types/defprotocol+ ISerializable
+  ;;; Serialization side
```

Comment:
what do you think if we prefix all methods for serialization with `ser` or `serialize`, and for deserialization : `des` or `deserialize` ?